src/javascript/binary/websocket_pages/user/account/settings/settings_detailsws.js

```diff
@@ -164,8 +164,9 @@ var SettingsDetailsWS = (function() {
             }
 
             // town/city
-            isRequiredError(fieldIDs.city);
-
+            if(!isRequiredError(fieldIDs.city) && !(/^[a-zA-Z0-9\s\,\.\-\/\(\)#']+$/).test(city)) {
```

Comment:
Nope, it should be isValidCity(fieldIDs.city) and isValidCity should be unit tested, and should be tested against all addresses already in DB to verify if it is correct or not. 🌴